Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java

```diff
@@ -184,6 +204,44 @@
 
     }
 
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.WORKSPACE, paramIndex = 2, userRole = { RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/rest/workspace/{workspaceid}/dictionaries.json", method = RequestMethod.GET, produces = "application/json")
+    @ResponseBody
+    public ResponseEntity<String> listWorkspaceDictionaryJson(
+            HttpServletRequest req,
+            @PathVariable("workspaceid") String workspaceId, Model model,
+            Principal principal) throws QuadrigaStorageException,
+            QuadrigaAccessException, QuadrigaException {
+        String userId = principal.getName();
+
+        List<IWorkspaceDictionary> dicitonaryList = null;
+        try {
+            dicitonaryList = workspaceDictionaryManager
+                    .listWorkspaceDictionary(workspaceId, userId);
+        } catch (QuadrigaStorageException e) {
+            logger.error("issue while adding dictionary ", e);
+        }
+        if (dicitonaryList == null) {
+            logger.info("Dictionar list is empty buddy");
+        }
+
+        JSONArray ja = new JSONArray();
+        for (IWorkspaceDictionary dictionary : dicitonaryList) {

```

Comment:
We are working on it. These pull requests were made earlier. Please don't look into it. We will create a new pull request.